### ISSUE TITLE
Support USB AUDIO in effects

### ISSUE THREAD

**Issue by dalb8** [carlosrafaelgn/FPlayAndroid#18]
The effects only support headset, headphones and loudspeaker but not audio via USB dongle.

**Comment by carlosrafaelgn** [comment]
Indeed! The problem is I do not own one of those dongles, so there is no way I could test it...

Moreover, the only references I could find regarding USB Audio were these (that work only on Android 23+ - Marshmallow+):
https://developer.android.com/reference/android/media/AudioDeviceInfo.html
https://developer.android.com/reference/android/media/AudioManager.html#getDevices(int)

Do you own one of those devices? If so, would you consider writing a code to detect usb audio, please? Then, I would merge your code into FPlay's code :)

**Comment by dalb8** [comment]
I own one but I'm not able to write code. I may be able to get another soon, in which case I'd give one to you. 

The AudioFX app in LineageOS as well as DSP Manager support USB audio.

**Comment by james34602** [comment]
Android USB audio sometimes quite buggy, I have a highly modified DSP Manager in my repository, my experience show some device may have conflict when using USB audio output, the system always mix up with analog output. So USB DAC can be output as analog output.
Whatever @carlosrafaelgn may experiment on USB audio :D

**Comment by dalb8** [comment]
OK the Loudspeaker tab changes the USB output, on my phone at least. It opened on loudspeaker tab when I opened the app with dongle plugged in but I didn't realise it would work until I read James' comment.

**Comment by carlosrafaelgn** [comment]
Precise detection of audio output is kinda buggy on Android... Check out methods Player._checkAudioSinkViaRoute() and Player._checkAudioSink() (lines 3318 and 3369).

They are a massive workaround, sadly, because different devices behave differently :(

Reading through the code in AudioSystem.java, I came across two constants:
public static final String DEVICE_OUT_USB_ACCESSORY_NAME = "usb_accessory";
public static final String DEVICE_OUT_USB_DEVICE_NAME = "usb_device";

Perhaps they could be used to try and detect whenever audio output is being done by the speaker or by an USB device???

**Comment by james34602** [comment]
I don't even know, have to do experiment on that. Carlos is right, different vendor may code audio framework differently...
